mantle/cmd/kola/testiso.go

```diff
@@ -794,6 +807,65 @@ func testLiveIso(ctx context.Context, inst platform.Install, outdir string, mini
 	return awaitCompletion(ctx, mach.QemuInst, outdir, completionChannel, mach.BootStartedErrorChannel, []string{liveOKSignal, signalCompleteString})
 }
 
+// testLiveFIPS verifies that adding fips=1 to the ISO results in a FIPS mode system
+func testLiveFIPS(ctx context.Context, outdir string) (time.Duration, error) {
+	tmpd, err := os.MkdirTemp("", "kola-testiso")
+	if err != nil {
+		return 0, err
+	}
+	defer os.RemoveAll(tmpd)
+
+	builddir := kola.CosaBuild.Dir
+	isopath := filepath.Join(builddir, kola.CosaBuild.Meta.BuildArtifacts.LiveIso.Path)
+	builder, config, err := newQemuBuilder(outdir)
+	if err != nil {
+		return 0, err
+	}
+	defer builder.Close()
+	if err := builder.AddIso(isopath, "", false); err != nil {
+		return 0, err
+	}
+
+	// This is the core change under test - adding the `fips=1` kernel argument via
+	// coreos-installer iso kargs modify should enter fips mode.
+	// Removing this line should cause this test to fail.
+	builder.AppendKernelArgs = "fips=1"
+
+	completionChannel, err := builder.VirtioChannelRead("testisocompletion")
+	if err != nil {
+		return 0, err
+	}
+
+	config.AddSystemdUnit("fips-verify.service", `
```

Comment:
Shouldn't we also check here that the crypto policy is correctly set to FIPS? (I.e. testing https://github.com/openshift/os/pull/1332.)

Reply:
Sure, done